DocCRender.docc/contributing/Internals.md

```diff
@@ -0,0 +1,189 @@
+
+```html
+<template>
+  <SVGIcon class="diagonal-arrow" viewBox="0 0 14 14">
+    <path d="M0.010 12.881l10.429-10.477-3.764 0.824-0.339-1.549 7.653-1.679-1.717 7.622-1.546-0.349 0.847-3.759-10.442 10.487z"></path>
+  </SVGIcon>
+</template>
+
+<script>
+import SVGIcon from '@/components/SVGIcon.vue';
+
+export default {
+  name: 'DiagonalArrowIcon',
+  components: { SVGIcon },
+};
+</script>
+``` 
+
+### Styles
+
+Styles in DocC Render are separated between `core` and `base` styles, with the majority of styling being defined as scoped inside the corresponding Vue file.
+
+> Define styles inside Vue files as **scoped**, so style collisions are reduced to a minimum. 
+
+* The `base` styles represent global styles, like resetting styles, styling links and headings across the app. These styles are imported once, none scoped, inside App.vue.
+* The `core` styles are used to define Sass variables, mixins and functions, that are to be used to generate reusable and complex styles inside Vue files. 
+
+> Important: SCSS files inside **core** should never output css directly! These style files are imported in almost every vue file, and making them output styles, will lead to duplication of styles across the entire app. 
+
+If you need access to a global mixin or var, from `_core.scss` you have to import it first.
+
+```html
+<style scoped lang="scss">
+    @import '@/styles/_core.scss';
+    .some-class {
+        margin-left: $varfrom-core-folder;
+    }
+</style>
+```
+
+## Unit tests
+
+DocC Render is unit tested, using the [Jest](https://jestjs.io/) testing framework. All components and utilities strive to have a corresponding test file inside the  `tests` folder, mirroring the file's location.
+
+Unit tests can be run via:
+
+* `npm run test:unit`  - run all unittests
+* `npm run test:unit:watch`  - run only the modified files and watch for changes
+* `npx run test:unit tests/unit/path/to/spec.js`  - run an individual test file
+
+## Code style
+
+DocC Render uses [ESLint](https://eslint.org/) to format the code and avoid syntax errors. It is following [Airbnb](https://github.com/airbnb/javascript) code styling along with [Vue Style Guide](https://vuejs.org/v2/style-guide/)’s essential configuration.
+
+To run the linter run:
+
+`npm run lint`
+
+To automatically fix any error, run:
+
+`npm run lint:fix`
+
+If you use the development server via `npm run serve`, files are linted on each change. Read more about configuring your linter [here](https://cli.vuejs.org/core-plugins/eslint.html#configuration)
+
+## Environment flags
+
+DocC Render has a few build-time environment flags, that allow you to set configuration parameters before building. Read [How to set Environment Variables in Vue CLI](https://cli.vuejs.org/guide/mode-and-env.html#environment-variables).
+
+**List of env variables:**
+
+* **VUE_APP_DEV_SERVER_PROXY** - The endpoint or path to read render JSON from
```

Comment:
```suggestion
* **VUE_APP_DEV_SERVER_PROXY** - The HTTP endpoint or  local filepath to read render JSON from when using the development server
```

I think it's important to call that out explicitly since the `/data` HTTP endpoint will always be used in production. This flag is just useful for local development purposes.